From 0feb74bd3acfa2b38773102347851a1d01314e83 Mon Sep 17 00:00:00 2001
From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Mon, 23 Nov 2020 22:27:27 +0530
Subject: [PATCH] update: Spoof signature to Google's for microG

* Needed to push updates through a F-Droid repo, since the APK
  would be signed with a different key than what we have the OS
  think
---
 fdroidserver/update.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/fdroidserver/update.py b/fdroidserver/update.py
index dd3bba50..062be000 100644
--- a/fdroidserver/update.py
+++ b/fdroidserver/update.py
@@ -1501,12 +1501,16 @@ def scan_apk(apk_file, require_signature=True):
     if require_signature:
         if not apk['sig']:
             raise BuildException(_("Failed to get APK signing key fingerprint"))
+        # Spoof signature for microG GmsCore and FakeStore
+        if apk['packageName'] == "com.google.android.gms" or apk['packageName'] == "com.android.vending":
+            logging.warning(_("Spoofing signature for {appid} with original signature {sig}")
+                        .format(appid=apk['packageName'], sig=apk['sig']))
+            apk['sig'] = 'd91564be0d13dcff260fe019eeea8cd8' # Google's signature
         apk['signer'] = common.apk_signer_fingerprint(
             os.path.join(os.getcwd(), apk_file)
         )
         if not apk.get('signer'):
             raise BuildException(_("Failed to get APK signing key fingerprint"))
-
     # Get size of the APK
     apk['size'] = os.path.getsize(apk_file)
 
-- 
2.46.0

